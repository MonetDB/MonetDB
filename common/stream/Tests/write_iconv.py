#!/usr/bin/env python3

from testdata import Doc, TestFile

import os
import subprocess
import sys
import tempfile


def run_streamcat(text, enc):
    content = bytes(text, 'utf-8')
    name = f'write_iconv_{enc}.txt'

    tf = TestFile(name, None)
    filename = tf.path()

    cmd = ['streamcat', 'write', filename, 'wstream', f'iconv:{enc}']

    print(f"Input UTF-8 encoded is {repr(content)}")
    # print(cmd)
    proc = subprocess.run(cmd, input=content)
    if proc.returncode != 0:
        print(f"{cmd} exited with status {proc.returncode}", file=sys.stderr)
        sys.exit(1)
    output = tf.read()
    os.remove(filename)
    print(f"Output is {repr(output)}")
    print()


text = "MøNëTDB"
run_streamcat(text, 'utf-8')
run_streamcat(text, 'latin1')
