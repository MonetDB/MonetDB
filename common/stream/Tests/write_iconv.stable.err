stderr of test 'write_iconv` in directory 'common/stream` itself:


# 15:02:03 >  
# 15:02:03 >  "/usr/bin/python3" "write_iconv.py" "write_iconv"
# 15:02:03 >  


# 15:02:03 >  
# 15:02:03 >  "Done."
# 15:02:03 >  

