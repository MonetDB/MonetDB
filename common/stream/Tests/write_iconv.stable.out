stdout of test 'write_iconv` in directory 'common/stream` itself:


# 15:02:03 >  
# 15:02:03 >  "/usr/bin/python3" "write_iconv.py" "write_iconv"
# 15:02:03 >  

Input UTF-8 encoded is b'M\xc3\xb8N\xc3\xabTDB'
Output is b'M\xc3\xb8N\xc3\xabTDB'

Input UTF-8 encoded is b'M\xc3\xb8N\xc3\xabTDB'
Output is b'M\xf8N\xebTDB'


# 15:02:03 >  
# 15:02:03 >  "Done."
# 15:02:03 >  

